core/src/test/scala/shapeless/serialization.scala

```diff
@@ -228,7 +234,7 @@ object SerializationTestDefns {
 
 }
 
-class SerializationTests {
+class SerializationTests extends Serializable {
   import SerializationTestDefns._
```

Comment:
This is only necessary in 2.12, not in 2.11... It seems that some materialized `Lazy` instances capture an outer reference... I have no idea why.